lib/Backend/arm64/LowerMD.cpp

```diff
@@ -8037,12 +8037,9 @@ LowererMD::CheckOverflowOnFloatToInt32(IR::Instr* instrInsert, IR::Opnd* intOpnd
 
     // CMP intOpnd, 0x7fffffff     -- Check for overflow
 
-    IR::RegOpnd *regOpnd= IR::RegOpnd::New(TyMachReg, this->m_func);
+    IR::RegOpnd *regOpnd= IR::RegOpnd::New(TyUint32, this->m_func);
 
-    instr = IR::Instr::New(Js::OpCode::MVN,
-        regOpnd,
-        IR::IntConstOpnd::New(0x80000000, TyInt32, this->m_func, true),
-        this->m_func);
+    instr = IR::Instr::New(Js::OpCode::MOVN, regOpnd, IR::IntConstOpnd::New(0x80000000, TyUint32, this->m_func, true), this->m_func);
     instrInsert->InsertBefore(instr);
```

Comment:
This can be collapsed into a CMN against 0x80000000, instead of allocating a reg and MVN the value into it.

Reply:
Actually, given the encoding support for SUBS (aka CMP), it looks like 0x80000000 can't be encoded directly either, so the legalizer will move that into a register.

Reply:
If we can allocate a temporary register, this does it in 3 instructions:
tmp = EOR_ASR31 src, src    ; tmp = src ^ ((int32)src >> 31) -- converts 0x80000000 into 0x7fffffff
EOR tmp, 0x7fffffff    ; this is encodeable
CBZ tmp, label    ; branch if tmp == 0x7fffffff

Reply:
An equivalent option, also 3 instructions:
tmp = EOR src, 0x80000000   ; gives 0 or -1 for overflow values
EOR_ASR31 tmp, tmp   ; tmp = tmp ^ ((int32)tmp >> 31) -- converts -1 to 0
CBZ tmp, label   ; branch if tmp was -1 or 0